pkg/kubelet/kuberuntime/kuberuntime_container_windows.go

```diff
@@ -77,5 +85,16 @@ func (m *kubeGenericRuntimeManager) generateWindowsContainerConfig(container *v1
 		wc.Resources.MemoryLimitInBytes = memoryLimit
 	}
 
-	return wc
+	// setup security context
+	effectiveSc := securitycontext.DetermineEffectiveSecurityContext(pod, container)
+	// RunAsUser only supports int64 from Kubernetes API, but Windows containers only support username.
+	// TODO: change RunAsUser to intstr in Kubernetes API.
```

Comment:
My preference is to keep the fields separate but open to feedback

Reply:
From CRI perspective, it supports both int64 and string for Linux containers but only string for Windows containers. While in today's API, only int64 is supported yet. So changing it to IntStr will work for both Linux and Windows.

Reply:
Ah, I see. So converging DetermineEffectiveSecurityContext will require the API change later. This PR just adds the new field run_as_user which is Windows only.

Reply:
exactly

Reply:
I think username could also be verified, e.g. administrator on Windows or root on Linux.

Reply:
If the docker images will run as user "foo" by default inside the container, (today) kubelet can't check what uid "foo" maps to, and whether it's actually root on the host.

How does this work for windows?...

Reply:
Privileged mode isn't implemented on Windows, and the container administrator account is separate because the filesystem and registry (where the user profile is) are separate from the host's. Putting a check on username won't add any security.

Reply:
If you do want it for consistency with Linux, it could be done but would need to check against localized variants of "Administrator" since [microsoft/windowsservercore](https://hub.docker.com/r/microsoft/windowsservercore/tags/) is localized

Reply:
> If the docker images will run as user "foo" by default inside the container, (today) kubelet can't check what uid "foo" maps to, and whether it's actually root on the host.

@yujuhong Good catch. Even non-root username could have uid 0 on Linux, so username won't make sense for linux security context.

> If you do want it for consistency with Linux, it could be done but would need to check against localized variants of "Administrator" since microsoft/windowsservercore is localized

@PatrickLang I think checking localized variants not possible because kubelet won't handle container localization. 

Let me remove the todo item.